lib/platform/bitbucket-server/index.js

```diff
@@ -121,13 +121,13 @@ async function initRepo({
   config = { projectKey, repositorySlug, gitPrivateKey };
 
   // Always gitFs
-  const { host } = url.parse(opts.endpoint);
+  const { host, pathname } = url.parse(opts.endpoint);
   const gitUrl = GitStorage.getUrl({
     gitFs: gitFs || 'https',
     auth: `${encodeURIComponent(opts.username)}:${encodeURIComponent(
       opts.password
     )}`,
-    host: `${host}/scm`,
+    host: `${host}${pathname}${pathname.endsWith('/') ? '' : '/'}scm`,
```

Comment:
the `pathname` is `'/'` if no path is in the URL
```
$ node
> url.parse('https://example.com').pathname
'/'
# matches with
> url.parse('https://example.com').href
'https://example.com/'
```

however
```
> url.parse('https://example.com/a/b').pathname
'/a/b'
```

so the `${pathname.endsWith('/') ? '' : '/'}` bit is to avoid a double slash in the event of something like `https://example.com/a/b/` (`pathname: '/a/b/'`)